docs/docs/guides/04-fe-coding-guidelines.mdx

```diff
@@ -0,0 +1,596 @@
+
+If you do mock a module try to create a manual mock instead under the `__mocks__` folder if
+possible.
+
+### Useful Articles
+
+- https://kentcdodds.com/blog/testing-implementation-details
+- https://testing-library.com/docs/queries/about/#priority
+
+## A11y
+
+Semantics are important for users, just because the application is “not on the web” does not mean
+our users are “fully abled” (do not have any type of disability). Therefore when building UI aim to
+use the correct semantic elements in the correct places, typically we use the `design-system` to
+build UI in the CMS, so most of the accessibility work is done there. But, accessibility is not easy
+– this is why we test from a user perspective as described in the “Testing” section. If you're not
+sure about accessibility patterns check out these resources:
+
+- https://www.w3.org/WAI/ARIA/apg/
+- https://webaim.org/
+- https://www.a11yproject.com/
+
+## Conventions
+
+### Submitting a PR
+
+When submitting an FE related PR ensure you've done the following:
+
+- Follow the template, especially including a clear description, it's a history for future
+  developers understanding the context around a decision.
+- Update or add tests for your relevant code, for more help on this consult the
+  [Testing Section](#unit-testing)
+- Always consider whether adding contributor documentation is helpful for the work you've done e.g.
+  you've created a new hook that can be used across the app
+- Get the relevant reviewers – engineers in your squad is a good start, but if you're working in a
+  space of the codebase you're not as familiar with e.g. `entity-service` consider getting engineers
+  who have more experience in the space.
+
+### Contributor Documentation
+
+:::caution
+No single squad “owns” the contributor documentation, it's a collective effort.
+:::
+
+The contributor docs are not “user” docs, they're related directly to the codebase for both internal
+and external engineers written with that in mind they explain technical concepts but also hold
+documentation for hooks, utils etc.
+
+#### Structure
+
+Currently the documentation is split into four sections:
+
+- Guides
+- Docs
+- API
+- RFCs
+
+The `Docs` section is organised to model the monorepo only to the package level e.g. `core > admin`
+past that point, they're organic and re-organise as we continue to grow the documentation. The `API`
+section at time of writing is light and only discusses the `Strapi` class. Moving forward, we'd like
+to include all the `admin` API endpoints and their return values.
+
+#### When should you add content?
+
+Content should be added typically when you add any new feature – especially enterprise e.g.
+`Review Workflows` and also when you add new code other engineers may find another use for e.g. a
+new hook. There's no guidelines on what to add vs what not to add, as defined above we add a variety
+of material to help onboard contributers.
```

Comment:
Should this content be in a more global space in the contrib docs and just linked to here? This content doesn't feel front-end specific.

Reply:
I agree.